bpf/http_sock.h

```diff
@@ -279,7 +279,9 @@ static __always_inline void finish_http(http_info_t *info, pid_connection_info_t
             bpf_ringbuf_submit(trace, get_flags());
         }
 
-        delete_server_trace();
+        if (info->type == EVENT_HTTP_REQUEST) {
```

Comment:
This was another bug, we were calling cleanup on the server traces even for client requests. So for Python applications (where everything happens on a single thread), if the server span made more than one client request we'd not propagate the trace id correctly on the second request.